tests/Features/ParentModelInheritsChildrenScopes.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Tightenco\Parental\Tests\Features;
+
+use Tightenco\Parental\Tests\Models\InternationalTrip;
+use Tightenco\Parental\Tests\Models\LocalTrip;
+use Tightenco\Parental\Tests\Models\Trip;
+use Tightenco\Parental\Tests\TestCase;
+
+class ParentModelInheritsChildrenScopes extends TestCase
+{
+    /** @test */
+    public function simple_scope_inheritance_check()
+    {
+        $this->createTenTrips();
+
+        $this->assertEquals(10, Trip::query()->count());
+        $this->assertCount(2, (new Trip)->getGlobalScopes());
+    }
+
+    /** @test */
+    public function global_scopes_can_be_added_on_the_fly()
+    {
+        $this->createTenTrips();
+
+        LocalTrip::addGlobalScope(function ($query) {
+            // this doesn't actually do anything to the query
+            $query->whereNotNull((new LocalTrip)->getInheritanceColumn());
+        });
+
+        $this->assertEquals(10, Trip::query()->count());
+        $this->assertCount(2, (new Trip)->getGlobalScopes());
+        $this->assertCount(3, (new LocalTrip)->getGlobalScopes());
+    }
+
+    /** @test */
+    public function child_scopes_apply_on_parent_queries()
+    {
+        $this->createTenTrips();
+
+        LocalTrip::addGlobalScope(function ($q) {
```

Comment:
I'm not sure adding child scopes to parent queries would be expected  behavior. I would expect parent scopes to be inherited by children, but not the other way around. Removing this constraint may simplify the implementation as well and make the scoping really easy.

We could check to see if the parent has any global scopes, if they do, just add them to the children in bootHasParent method. What do you think? Am I missing something that would make that hard?

Reply:
Well, the child already inherits the global scopes from the parent if they're in the boot method, now I doubt the children will inherit "on the fly" global scopes - that's not what I was trying to do in this PR, but I think it can be done by overriding the addGlobalScope method in the same way as the [regiserModelEvent method](https://github.com/tightenco/parental/blob/master/src/HasChildren.php#L11).

Let's imagine the following use-case where inheriting the child global scopes on the parent would be useful (the global scopes would ONLY apply to the right child type btw):
- We've got a parent PaymentMethod model, with a few child models such as CreditCard, BankAccount, w/e.
- We add a global scope (NotExpiredCard) on the CreditCard to avoid querying expired cards.
- When querying all payment methods, we expect (or I do, at least ^^) to only get valid payment methods, but we'll get expired credit cards as well.
- That's precisely the aim of this PR, it prevents expired cards from being returned, but idk if that's what you'd want in this package.
- The NotExpiredCard scope will only apply on credit cards, it won't apply to bank accounts, or other payment methods.

Reply:
Ah, thank you for writing that out so clearly. That makes perfect sense. And I believe you're right. That IS how I would expect the package to behave. So I guess the question now is: how do we handle the scoping without something complex like a separate file. I haven't really rolled my sleeves up on this problem, but I'd like to soon to see what I come up with. If you have any other ideas, they'd be great.